src/lib.rs

```diff
@@ -191,21 +291,32 @@ pub mod pallet {
 		pub fn show_me_the_money(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 			let payee = ensure_signed(origin)?;
 
-			// Calculate the veted amount on demand.
+			// Calculate the vested amount on demand.
 			let mut info =
 				AccountsPayable::<T>::get(&payee).ok_or(Error::<T>::NoAssociatedClaim)?;
 			ensure!(
 				info.claimed_reward < info.total_reward,
 				Error::<T>::RewardsAlreadyClaimed
 			);
-			let now = frame_system::Pallet::<T>::block_number();
 
-			let payable_per_block = info.total_reward
+			ensure!(
+				info.total_reward > T::MinimumContribution::get(),
+				Error::<T>::ContributionNotHighEnough
+			);
+
+			// Vesting is done in relation with the relay chain slot
+			let now: T::BlockNumber = RelayChainBeacon::<T>::slot().into();
+
+			// Substract the first payment from the vested amount
+			let first_paid = T::InitializationPayment::get() * info.total_reward;
+
+			let payable_per_block = (info.total_reward - first_paid)
 				/ T::VestingPeriod::get()
 					.saturated_into::<u128>()
 					.try_into()
 					.ok()
 					.ok_or(Error::<T>::WrongConversionU128ToBalance)?; //TODO safe math;
+
```

Comment:
The TODO above seems really important, as this is truncating integer division (right?)